pkg/client/cache/delta_fifo.go

```diff
@@ -118,6 +118,10 @@ type DeltaFIFO struct {
 	// purpose of figuring out which items have been deleted
 	// when Replace() or Delete() is called.
 	knownObjects KeyListerGetter
+
+	// indication the queue is closed
+	closed bool
```

Comment:
Can you add comments about why this isn't used to gate adds?  I'm assuming its because you don't consider it worth the lock acquisition, but I think its worth an explicit comment.